zcash_client_sqlite/src/chain.rs

```diff
@@ -64,6 +76,148 @@ where
     Ok(())
 }
 
+/// Data structure representing a row in the block metadata database.
+#[cfg(feature = "unstable")]
+pub struct BlockMeta {
+    pub height: BlockHeight,
+    pub block_hash: BlockHash,
+    pub block_time: u32,
+    pub sapling_outputs_count: u32,
+    pub orchard_actions_count: u32,
+}
+
+#[cfg(feature = "unstable")]
+impl BlockMeta {
+    pub fn block_file_path<P: AsRef<Path>>(&self, blocks_dir: &P) -> PathBuf {
+        blocks_dir.as_ref().join(Path::new(&format!(
+            "compactblock-{}-{}",
```

Comment:
Ideally humans aren't going to be looking at this. Even so, I think that grouping by type is more valuable than grouping by block, particularly since when we start doing out-of-order operations, it will be commonplace to have some parts for some blocks and not others, and in that case one will likely be grepping the output of `ls` for a particular block height or block hash anyway.